packages/protocol/test-sol/integration/Integration.t.sol

```diff
@@ -1,19 +1,81 @@
 // SPDX-License-Identifier: UNLICENSED
-pragma solidity ^0.5.13;
+pragma solidity >=0.5.13 <0.8.20;
 
 import "celo-foundry/Test.sol";
-import "@celo-contracts/common/GoldToken.sol";
-// import "@celo-contracts/common/test/MockGoldToken.sol";
-import "forge-std/console.sol";
+
+import { Utils } from "@test-sol/utils.sol";
+import { Constants } from "@test-sol/constants.sol";
+
 import "@celo-contracts/common/interfaces/IRegistry.sol";
+import "@celo-contracts/common/interfaces/IProxy.sol";
 
 contract IntegrationTest is Test {
   address constant registryAddress = address(0x000000000000000000000000000000000000ce10);
-  address account1 = actor("account1");
-  address account2 = actor("account2");
   IRegistry registry = IRegistry(registryAddress);
 
   function setUp() public {}
+}
+
+contract RegistryIntegrationTest is IntegrationTest, Utils, Constants {
+  IProxy proxy;
+
+  function test_shouldHaveAddressInRegistry() public view {
+    for (uint256 i = 0; i < contractsInRegistry.length; i++) {
+      string memory contractName = contractsInRegistry[i];
+      address contractAddress = registry.getAddressFor(keccak256(abi.encodePacked(contractName)));
+      console2.log(contractName, "address in Registry is: ", contractAddress);
+      assert(contractAddress != address(0));
+    }
+  }
+
+  function test_shouldHaveCorrectBytecode() public {
+    for (uint256 i = 0; i < contractsInRegistry.length; i++) {
+      // Read name from list of core contracts
+      string memory contractName = contractsInRegistry[i];
+      console2.log("Checking bytecode of:", contractName);
+
+      // Converting contract names to hashes for comparison
+      bytes32 hashContractName = keccak256(abi.encodePacked(contractName));
+      bytes32 hashAccount = keccak256(abi.encodePacked("Accounts"));
+      bytes32 hashElection = keccak256(abi.encodePacked("Election"));
```

Comment:
Since these are constants we can move them outside of cycle

Reply:
Good point, that's done in https://github.com/celo-org/celo-monorepo/pull/11002/commits/7a740cbbecd1d607ba9caa0f6882ceb8a3a558b4.